s3/s3_transport/include/s3_transport.hpp

```diff
@@ -0,0 +1,1901 @@
+                            // file may not exist, open with std::ios_base::out to create then with in/out
+                            cache_fstream_.open(cache_file_path_.c_str(), mode | std::ios_base::out);
+                            cache_fstream_.close();
+                            cache_fstream_.open(cache_file_path_.c_str(), mode | std::ios_base::in | std::ios_base::out);
+                        } else {
+                            rodsLog(config_.debug_log_level, "%s:%d (%s) [[%lu]] opened cache file [trunc_flag=%d]\n", __FILE__, __LINE__, __FUNCTION__, get_thread_identifier(), trunc_flag);
+                        }
+
+                        if (!cache_fstream_ || !cache_fstream_.is_open()) {
+                            rodsLog(LOG_ERROR, "%s:%d (%s) [[%lu]] Failed to open cache file %s\n",
+                                    __FILE__, __LINE__, __FUNCTION__, this->get_thread_identifier(), cache_file_path_.c_str());
+                            this->critical_error_encountered_ = true;
+                            return_value = false;
+                        }
+
+                        if (!this->seek_to_end_if_required(this->mode_)) {
+                            critical_error_encountered_ = true;
+                            return_value = false;
+                        }
+                    }
+
+                } else {
+
+                    // not using cache, just create our own fd
+
+                    const auto fd = this->file_descriptor_counter_++;
+
+                    if (fd < minimum_valid_file_descriptor) {
+                        this->critical_error_encountered_ = true;
+                        return_value = false;
+                    }
+
+                    this->fd_ = fd;
+
+                    if (!this->seek_to_end_if_required(mode_)) {
+                        this->critical_error_encountered_ = true;
+                        return_value = false;
+                    }
+                }
+
+                // set the number of threads to close to the number of client transfer threads
+                // this will count down as closes happen
+                if (data.threads_remaining_to_close == 0) {
+                    data.threads_remaining_to_close = this->config_.number_of_client_transfer_threads;
+                }
+
+
+            });
+
+            return return_value;
+
+
+        }  // end open_impl
+
+        error_codes initiate_multipart_upload()
+        {
+            namespace bi = boost::interprocess;
+            namespace types = shared_data::interprocess_types;
+
+            //int cache_fd = -1;
+            unsigned int retry_cnt    = 0;
+            //bool enable_md5 = config_.enable_md5_flag;
+            put_props_.useServerSideEncryption = config_.server_encrypt_flag;
+            std::stringstream msg;
+
+            put_props_.md5 = nullptr;
+            //if ( enable_md5 )
+            //    putProps.md5 = s3CalcMD5( cache_fd, 0, object_size, resource_name() );
```

Comment:
dead

Reply:
Removed although I need to look into whether doing an MD5 calculation if feasible for streaming.

Adding in a comment  to the README that this configuration is not currently implemented .

Reply:
Added an issue #1947 for this.